### PR TITLE
Refactor p2p unit tests

### PR DESCRIPTION
**Pull request by joshua-kim**
## Why this should be merged

the P2P unit tests are confusing + spin up goroutines, this PR refactors all of them to be simpler

## How this works

See above

## How this was tested

UTs

### CODE REVIEW

File: plugin/evm/vm.go [hunk 1]
```diff
@@ -343,6 +343,12 @@ type VM struct {
 	// Avalanche Warp Messaging backend
 	// Used to serve BLS signatures of warp messages over RPC
 	warpBackend warp.Backend
+
+	p2pSender             p2p.AppSender
+	ethTxGossipHandler    p2p.Handler
+	atomicTxGossipHandler p2p.Handler
+	ethTxGossiper         gossip.Gossiper
+	atomicTxGossiper      gossip.Gossiper
```
Comment (joshua-kim) [inline plugin/evm/vm.go:351]: Keeping these interfaces here lets us override these with noop implementations during testing.

File: scripts/versions.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
 
 # Don't export them as they're used in the context of other calls
-avalanche_version=${AVALANCHE_VERSION:-'v1.10.18-rc.0'}
+avalanche_version=${AVALANCHE_VERSION:-'sender-refactor'}
```
Comment (darioush) [inline scripts/versions.sh:4]: is this necessary? (only impacts e2e tests)

File: plugin/evm/vm.go (hunk 1 above)
Comment (darioush) [reply plugin/evm/vm.go:351]: could we add this as a comment?

**Review by darioush — DISMISSED** [review]
I think this is a good improvement so I am approving this PR.

I think we should find a coherent strategy for initializing & wiring objects during testing. May be worthwhile to look into golang best practices or agree on some standards for further improvement.

File: scripts/versions.sh (hunk 2 above)
Comment (joshua-kim) [reply scripts/versions.sh:4]: I'm not sure how to get e2e working with a custom tag that isn't an official version tag... haven't dove to see why it doesn't work

File: peer/network_test.go [hunk 3]
```diff
@@ -425,6 +425,7 @@ func TestRequestMinVersion(t *testing.T) {
 	responseBytes, _, err := client.SendAppRequestAny(
 		context.Background(),
 		&version.Application{
+			Name:  "avalanche",
```
Comment (joshua-kim) [inline peer/network_test.go:428]: Diff included because of the upgrade to latest avalanchego

File: plugin/evm/vm.go [hunk 4]
```diff
@@ -344,7 +344,8 @@ type VM struct {
 	// Used to serve BLS signatures of warp messages over RPC
 	warpBackend warp.Backend
 
-	p2pSender             p2p.AppSender
+	// Can be overridden in unit tests
```
Comment (darioush) [inline plugin/evm/vm.go:347]:
```suggestion
	// Initialize only sets these if nil so they can be overridden in tests
```

**Review by darioush — APPROVED** [review]

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**